crates/bifrost/src/watchdog.rs

```diff
@@ -71,39 +82,79 @@ impl Watchdog {
                 log_id,
                 requested_trim_point,
             } => {
-                self.on_log_trim(log_id, requested_trim_point);
+                self.store_trim_request(log_id, requested_trim_point);
             }
         }
     }
 
-    fn on_log_trim(&self, log_id: LogId, requested_trim_point: Lsn) {
+    fn spawn_trim(&self, mut trim_requests: TrimRequests) -> Result<TaskHandle<()>, ShutdownError> {
         let bifrost = self.inner.clone();
-        let _ = TaskCenter::spawn_child(
+        TaskCenter::spawn_unmanaged(
             TaskKind::BifrostBackgroundLowPriority,
-            format!("trim-chain-{log_id}"),
+            "trim-chains",
             async move {
-                let trim_point = bifrost
-                    .get_trim_point(log_id)
-                    .await
-                    .context("cannot determine tail")?;
-                if trim_point == Lsn::INVALID {
-                    return Ok(());
+                if trim_requests.is_empty() {
+                    return;
+                }
+
+                let mut trim_points = Vec::with_capacity(trim_requests.len());
+
+                // Concurrently look up the trim points of all the requests
+                let mut trim_point_futures: FuturesUnordered<_> = trim_requests
+                    .drain()
+                    .map(async |(log_id, requested_trim_point)| {
+                        match bifrost.get_trim_point(log_id).await {
+                            // an invalid lsn means that the log has never been trimmed
+                            Ok(actual_trim_point) if actual_trim_point == Lsn::INVALID => Ok(None),
+                            Ok(actual_trim_point) => Ok(Some(TrimPoint {
+                                log_id,
+                                requested_trim_point,
+                                actual_trim_point,
+                            })),
+                            Err(_) => Err((log_id, requested_trim_point)),
+                        }
+                    })
+                    .collect();
+
+                while let Some(trim_point_result) = trim_point_futures.next().await {
+                    match trim_point_result {
+                        Ok(Some(trim_point)) => {
+                            trim_points.push(trim_point);
+                        }
+                        // skip invalid lsns
+                        Ok(None) => continue,
+                        Err((log_id, requested_trim_point)) => {
+                            // keep track of requests we failed to find trim points for
+                            warn!(
+                                "Bifrost watchdog failed to find a trim point for {log_id}; will not be able to process the request to trim to to {requested_trim_point}"
```

Comment:
Minor typo; since we log an overall warning at the end of the cycle, I wonder if this shouldn't be `debug!` rather - as it would one per failed log, this could get unnecessarily noisy.

```suggestion
                                "Bifrost watchdog failed to find a trim point for {log_id}; will not be able to process the request to trim chain to {requested_trim_point}"
```